gen/field.go

```diff
@@ -461,6 +465,46 @@ func (f fieldGroupGenerator) Equals(g Generator) error {
 		`, f)
 }
 
+func (f fieldGroupGenerator) Zap(g Generator) error {
+	return g.DeclareFromTemplate(
+		`
+		<$zapcore := import "go.uber.org/zap/zapcore">
+		<$v := newVar "v">
+		<$enc := newVar "enc">
+
+		// MarshalLogObject implements zapcore.ObjectMarshaler, enabling
+		// fast logging of <.Name>.
+		func (<$v> *<.Name>) MarshalLogObject(<$enc> <$zapcore>.ObjectEncoder) error {
+			<range .Fields>
+				<- $fname := goName . ->
+				<- $fval := printf "%s.%s" $v $fname ->
+				<- if .Required ->
+					<- $encAdd := printf "%s.Add%s(%q, %s)" $enc (zapEncoder .Type) .Name (zapMarshaler .Type $fval) ->
+					<if (zapCanError .Type) ->
+						if err := <$encAdd>; err != nil {
+							return err
+						}
+					<- else ->
+						<$encAdd>
+					<- end ->
+				<- else ->
+					<- $encAdd := printf "%s.Add%s(%q, %s)" $enc (zapEncoder .Type) .Name (zapMarshalerPtr .Type $fval) ->
+					if <$fval> != nil {
+						<if (zapCanError .Type) ->
+							if err := <$encAdd>; err != nil {
+								return err
+							}
+						<- else ->
+							<$encAdd>
+						<- end ->
+					}
+				<- end>
+			<end>
```

Comment:
nit: use `<end ->` for to end the range so that if .Fields is empty, we don't get an empty line.